src/components/Message/hooks/useHandleMute.js

```diff
@@ -0,0 +1,89 @@
+// @ts-check
+import { useContext } from 'react';
+import { isUserMuted, validateAndGetMessage } from '../utils';
+import { ChannelContext, TranslationContext } from '../../../context';
+
+/**
+ * @typedef {{
+ *   notify?: import('types').MessageComponentProps['addNotification'],
+ *   getSuccessNotification?: import('types').MessageComponentProps['getMuteUserSuccessNotification'],
+ *   getErrorNotification?: import('types').MessageComponentProps['getMuteUserErrorNotification'],
+ * }} NotificationArg
+ * @type {(message: import('stream-chat').MessageResponse, notification: NotificationArg) => (event: React.MouseEvent<HTMLElement>) => Promise<void>}
+ */
+export const useHandleMute = (message, notifications) => {
```

Comment:
maybe `useMuteHandler` makes more sense? What do you think?

Reply:
Makes total sense 👍